alphagov/support
# Changes to the 'Request a new campaign' form

## Background

We were asked to implement changes to the 'Request a new campaign' form.

[Trello](https://trello.com/c/Zx0Eh1Ch/3130-changes-to-the-request-a-new-campaign-form-2)

## What

- Remove the checkboxes: 
  - “Have you read the GCS guidance on campaign websites and accepted the requirements for a Campaign Platform website?” 
  - “Have you followed the GCS guidance for OASIS planning and are using the mandatory GCS OASIS template?”

- Add the following checkboxes in their place:

  - I/We have read the[ GCS guidance on campaign websites](https://gcs.civilservice.gov.uk/guidance/campaigns/websites-government-campaigns/) and accept the requirements for a Campaign Platform website

  - I/We have followed the [GCS guidance for OASIS planning](https://gcs.civilservice.gov.uk/wp-content/uploads/2015/09/6.3938_CO_GCS-Campaign-Planning_FINAL_A4_111017.pdf) and are using the [mandatory GCS OASIS template](https://gcs.civilservice.gov.uk/guidance/marketing/delivering-government-campaigns/guide-to-campaign-planning-oasis/)

  - I/We will take full responsibility for all aspects of managing and resourcing this campaign site from Discovery stage to site build/content development, to site closure. The site must be regularly reviewed to ensure a high standard of work. GCS reserves the right to close the site if standard of the site drops below expectations.

  - I/We will ensure the site meets all[ government web accessibility standards](https://www.gov.uk/guidance/accessibility-requirements-for-public-sector-websites-and-apps), and that it will be tested and the Accessibility Statement completed before final review (if your team doesn't have expertise in accessible content design then appropriate training must be undertaken before the site build.) 
  - I/We agree to take responsibility to maintain and up-date the Cookie Notice and Privacy Notice as necessary, with our Data Protection Officer (NB : GDS will add a basic "boiler plate", however departments will need to identify if additions are needed).

- Ensure that these checkbox values are passed through to Zendesk on-submit
- Copy changes in to the form as per https://docs.google.com/document/d/1XAh4uKaJ-TnV-FZ6IDOknwgyfanvakdo/edit

| Part 1 Before  | 
| ------------- |
| <img width="580" alt="Screenshot 2023-04-04 at 17 50 55" src="https://user-images.githubusercontent.com/38078064/229862114-069a2657-9c42-4b36-a269-9b57bec05e21.png"> | 

| Part 1 After  | 
| ------------- |
| <img width="880" alt="Screenshot 2023-04-04 at 17 39 26" src="https://user-images.githubusercontent.com/38078064/229861766-b02d5db8-b4f3-4be0-9131-2c7db58f8cf5.png"> |


| Part 2 Before  | 
| ------------- |
| <img width="780" alt="Screenshot 2023-04-04 at 17 51 07" src="https://user-images.githubusercontent.com/38078064/229862199-8fd30ef3-f86f-40cf-8b14-f47d00c49cf8.png"> | 

| Part 2 After  | 
| ------------- |
| <img width="877" alt="Screenshot 2023-04-04 at 16 44 12" src="https://user-images.githubusercontent.com/38078064/229861864-b5e7425c-95eb-4723-8767-0521d50b8e41.png"> |

- Checkboxes validation
<kbd>
<img width="930" alt="Screenshot 2023-04-04 at 16 07 09" src="https://user-images.githubusercontent.com/38078064/229862558-8561478e-dae8-45f1-a5e1-d07641ee2c71.png">
</kbd>

## How to test it
- visit `/campaign_request/new` path
- try to submit the form

---
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.
